Src/libCZI/stdAllocator.cpp

```diff
@@ -30,24 +30,27 @@ void* CHeapAllocator::Allocate(std::uint64_t size)
 	{
 		throw std::out_of_range("The requested size for allocation is out-of-range.");
 	}
-#if defined(__EMSCRIPTEN__)||defined(__APPLE__)
-	return malloc((size_t)size);
-#else
-#if defined(__GNUC__)
+	// it's not clear to me why the code needs the returned pointer to be at a memory boundary
```

Comment:
I would not submit to libCZI with this comment.  It would be ideal to completely understand the code change you are submitting.

Reply:
For the sake of discussion:
I'm not an expert on this but here is my limited understanding:
* the cpu fetches memory from RAM into local caches at run time
* so when you need to access a big data structure, the cpu actually delivers it in chunks in the local cache
* the local cache is small but super fast to access
* so it is a net gain to move chunks mem into local cache at run time rather than access the big blob in global ram
* the local cache speed is super fast
* so if your data structure has an element half in and half out of the cache, then you will pay a cost while iterating
* it will have to evict cache and load another chunk
* which is why you want your structures to align to cache boundaries